### PR TITLE
[STREAMPIPES-616] refactor cumsum processor

### PR DESCRIPTION
**Pull request by bossenti**
Co-authored-by: smlabt <git@samuelabt.de>

  
  

### Purpose

Collaboratively with @smlabt, I've refactored cusum processor from the old three-class-based approach to the new one-class-based approach for designing processing elements.

### Approach

We marked the three abstract class that have been used for describing processing elements as deprecated from the current version on. Additionally, we decided to rename the refactored processing element to better reflect it's intention.
For backwards compatibility, we decided to keep the old processing element and mark it as deprecated too. So we can remove it after one or two releases.

### Remarks

Fixes: STREAMPIPES-616
We will use STREAMPIPES-615 to refactor the existing processing elements step by step.

### CODE REVIEW

File: streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.cusum/strings.en [hunk 1]
```diff
@@ -1,5 +1,5 @@
-org.apache.streampipes.processors.changedetection.jvm.cusum.title=Cusum
-org.apache.streampipes.processors.changedetection.jvm.cusum.description=
+org.apache.streampipes.processors.changedetection.jvm.cusum.title=Cusum (Deprecated)
+org.apache.streampipes.processors.changedetection.jvm.cusum.description=This processor is deprecated please, use the processing element `WelfordChangeDetection` instead.
```
Comment (grainier-macrometa) [inline streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.cusum/strings.en:2]:
```suggestion
org.apache.streampipes.processors.changedetection.jvm.cusum.description=This processor is deprecated.​ ​Please, use the processing element `WelfordChangeDetection` instead.
```

File: streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/java/org/apache/streampipes/processors/changedetection/jvm/welford/WelfordChangeDetection.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    @Override
+    public void onEvent(Event event, SpOutputCollector collector) throws SpRuntimeException {
+
+        Double number = event.getFieldBySelector(selectedNumberMapping).getAsPrimitive().getAsDouble();
+        welfordAggregate.update(number);  // update mean and standard deviation
+        Double normalized = getZScoreNormalizedValue(number);
+        updateStatistics(normalized);
+
```
Comment (grainier) [inline streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/java/org/apache/streampipes/processors/changedetection/jvm/welford/WelfordChangeDetection.java:99]:
```suggestion
```

File: streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.cusum/documentation.md [hunk 3]
```diff
@@ -28,6 +28,8 @@
 
 ## Description
 
+**This processing element is deprecated please use the `WelfordChangeDetection` processing element instead.**
```
Comment (grainier) [inline streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.cusum/documentation.md:31]:
```suggestion
**This processing element is deprecated. Please use the `WelfordChangeDetection` processing element instead.**
```

File: streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.cusum/strings.en (hunk 1 above)
Comment (grainier) [inline streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.cusum/strings.en:2]:
```suggestion
org.apache.streampipes.processors.changedetection.jvm.cusum.description=This processor is deprecated. Please use the processing element `WelfordChangeDetection` instead.
```

File: streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.welford/strings.en [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+org.apache.streampipes.processors.changedetection.jvm.welford.description=Change detection for a time series based on Welford's algorithm.
+
+number-mapping.title=Value to observe
+number-mapping.description=Specifies the monitored dimension.
+
+param-k.title=Parameter k
+param-k.description=The sensitivity parameter. High value indicates low sensitivity. Unit: Standard deviations
+
+param-h.title=Parameter h
+param-h.description=The threshold above which a change is detected. Unit: Standard deviations
```
Comment (grainier) [inline streampipes-extensions/streampipes-processors-change-detection-jvm/src/main/resources/org.apache.streampipes.processors.changedetection.jvm.welford/strings.en:11]:
```suggestion
param-h.description=The threshold above which a change is detected. Unit: Standard deviations

```

**Comment by bossenti** [comment]
thanks a lot @grainier!

**Comment by tenthe** [comment]
Hey @smlabt & @bossenti thanks a lot for refactoring the processor

**Pull request merged**